pkg/manager/member/pd_scaler.go

```diff
@@ -101,7 +101,26 @@ func (psd *pdScaler) ScaleIn(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet,
 		return fmt.Errorf("TidbCluster: %s/%s's pd status sync failed,can't scale in now", ns, tcName)
 	}
 
-	err := controller.GetPDClient(psd.pdControl, tc).DeleteMember(memberName)
+	pdClient := controller.GetPDClient(psd.pdControl, tc)
+	// If the pd pod was pd leader during scale-in, we would transfer pd leader to pd-0 directly
+	// If the pd statefulSet would be scale-in to zero and the pd-0 was going to be deleted,
+	// we would directly deleted the pd-0 without pd leader transferring
+	if ordinal > 0 {
+		leader, err := pdClient.GetPDLeader()
+		if err != nil {
+			return err
+		}
+		if leader.Name == memberName {
+			err = pdClient.TransferPDLeader(fmt.Sprintf("%s-pd-%d", tc.GetName(), 0))
```

Comment:
1) this is used to transfer lead to the first pod (pod-0), but pod-0 maybe not working at the time of transferring 
2) does it take effect on the completion of this request? if not, the problem still exist

what do you think about this solution?

- delete current PD member (same as before)
- check if it is leader (if not return nil, if yes return error and let the controller to retry next time)

Reply:
> 1. this is used to transfer lead to the first pod (pod-0), but pod-0 maybe not working at the time of transferring
> 2. does it take effect on the completion of this request? if not, the problem still exist
> 
> what do you think about this solution?
> 
> * delete current PD member (same as before)
> * check if it is leader (if not return nil, if yes return error and let the controller to retry next time)

Currently PD upgrading would transfer pd leader to the first and the last ordinal pd pod without considering whether both of them were available. I think it is necessary to check whether the target pd pod is working before transferring pd leader, maybe it should be fixed in another issue.

In my solution, if the target pd pod was the pd leader, it would be transferring pd leader to pd-0 requeue this request to wait for next round.

As for your solution, I don't understand how can the target pod ( which was also pd leader ) transferring pd leader If we only return nil.

Reply:
when the PD member is deleted and after all members receive this message, it will not be the leader anymore (in the background, the PD cluster will elect a new leader).

so, we can simply delete it and wait for it not be a leader (if it was a leader).

Reply:
If we scale-in pd cluster from 5 to 3, it could be at most 2 times pd leader tranferring. First pd-4 to pd-3 , and Second pd-3 to pd -2, which I think would be high risky.